### PR TITLE
add visit exception to know on which component any exception occured

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: sonar-testing-harness/src/main/java/org/sonar/test/ExceptionCauseMatcher.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  public ExceptionCauseMatcher andNoMessage() {
+    return new ExceptionCauseMatcher(type, EXPECT_NO_MESSAGE_CONSTANT);
+  }
+
+  @Override
+  protected boolean matchesSafely(Throwable item) {
+    return item.getClass().isAssignableFrom(type)
+      && (expectedMessage == null || EXPECT_NO_MESSAGE_CONSTANT.equals(expectedMessage) ? item.getMessage() == null : item.getMessage().contains(expectedMessage));
```
Comment (SonarTech) [inline sonar-testing-harness/src/main/java/org/sonar/test/ExceptionCauseMatcher.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)

File: server/sonar-server/src/main/java/org/sonar/server/computation/component/PathAwareCrawler.java [hunk 2]
```diff
@@ -123,4 +139,41 @@ private T createForComponent(Component component) {
...
+
+    private final DequeBasedPath<T> currentPath;
+
+    private ComponentPathPrinter(DequeBasedPath<T> currentPath) {
+      this.currentPath = currentPath;
+    }
+
+    @Override
+    public String toString() {
+      return currentPath.isRoot() ? "" : " located " + toKeyPath(currentPath);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/component/PathAwareCrawler.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
Please apply the format of the log, and I also think that displaying the path in PathAwareCrawler is useless.

**Pull request merged**